content/backend/typescript-apollo/1-getting-started.md

```diff
@@ -0,0 +1,464 @@
+
+```graphql(path=".../hackernews-typescript/schema.graphql"&nocopy)
+type Query {
+  ok: Boolean!
+}
+```
+
+The `nexus-typegen.ts` contains a lot of autogenerated typescript `interfaces` and `types`. You won't have to dig too deep into these, as these will be added automatically to your nexus function signatures. Though at certain times, you might look up a type from here and use it manually in your code. 
+
+Before you move on to the next section, you will add a `generate` script to your `package.json` so you can quickly regenerate your schema at any time. You will also add another `dev` script, which will come in handy in the next section to start our web server. 
+
+<Instruction>
+
+Add the `generate` and `dev` script to your package.json
+
+```json{3-4}(path=".../hackernews-typescript/package.json")
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "dev": "ts-node-dev --transpile-only --no-notify --exit-child src/index.ts",
+    "generate": "ts-node --transpile-only src/schema.ts"
+  },
+```
+</Instruction>
+
+1. You can run `npm run generate` to update your `schema.graphql` and `nexus-typegen.ts` file when there are any changes in your `nexus` code.
+2. You can use `npm run dev` to start the web server and watch for any changes. 
+
+### Creating a GraphQL server
+
+Now that you have a schema, you can finally create your GraphQl server inside `index.ts`. 
+
+<Instruction>
+
+Write up the following code to start the server in `src/index.ts`. 
+
+```typescript(path=".../hackernews-typescript/src/index.ts")
+import { ApolloServer } from "apollo-server";
+
+// 1
+import { schema } from "./schema";
+export const server = new ApolloServer({
+    schema,
+});
+
+const port = 3000;
+// 2
+server.listen({port}).then(({ url }) => {
+    console.log(`🚀  Server ready at ${url}`);
+});
+
+```
+
+</Instruction>
+
+Let's understand what's going on here by walking through the numbered comments: 
+
+1. The schema object you created using `nexus` defines your GraphQL schema. You need to provide this when instantiating your server.
+
+2. You start the server and specify the port. After the server starts, it returns a `url` string inside a promise. 
+
+Now that your server is ready let's give it a run and see what happens!
+
+<Instruction> 
+
+Open up your terminal and run the following command
+
+```bash(path=".../hackernews-typescript/")
+npm run dev
+```
+
+</Instruction>
+
+If everything goes well, you should see the following output inside the terminal:
+
+
+```(nocopy)
+🚀  Server ready at http://localhost:3000/
+```
+
+Now if you navigate to `http://localhost:3000/` you should see the following page: 
+
+![Apollo Server Landing Page](https://i.imgur.com/2bOq8c1.png)
+
+If you click on the `Query your Server` button, you will be redirected to The Apollo Studio Explorer. This is an _online web-based GraphQL IDE_ for running queries and exploring your schema (along with many other advanced features). 
```

Comment:
When referencing UI elements in a text, I've come to adopt a consistent technique of always making the text of the UI element bold.

The rational is that it stands out of the current text and gives a strong visual anchor to the UI element. While I think that with consistent usage, inline code would also work, I prefer using inline code formatting for actual code, file names, CLI commands, ...

If you want to see an example of where I applied these bold visual anchors quite a bit, see this tutorial: https://dev.to/prisma/how-to-setup-a-free-postgresql-database-on-heroku-1dc1#:~:text=Heroku%20offers%20a%20free%20plan,database%22%20for%20experimentation%20or%20prototyping.

```suggestion
If you click on the **Query your server** button, you will be redirected to The Apollo Studio Explorer. This is an _online web-based GraphQL IDE_ for running queries and exploring your schema (along with many other advanced features). 
```

Reply:
```suggestion
If you click on the `Query your Server` button, you will be redirected to the [Apollo Studio Explorer](https://www.apollographql.com/docs/studio/explorer/). This is an _online web-based GraphQL IDE_ for running queries and exploring your schema (along with many other advanced features). 
```